Sources/ApolloCodegenLib/Templates/RenderingHelpers/OperationTemplateRenderer.swift

```diff
@@ -50,4 +50,40 @@ extension OperationTemplateRenderer {
       """
   }
 
+  func DeferredProperties(
+    _ definition: IR.Definition
+  ) -> TemplateString {
+    func isDeferred(_ selectionSet: IR.SelectionSet) -> Bool {
```

Comment:
@AnthonyMDev - needing your feedback here whether this is the best way to be determining at an operation-level whether any fragments are deferred; I think it should be since we're only doing direct selections.

Reply:
I think this code should be valid, but I'm wondering if we could prevent another iteration through the entire selection set here.

What if we added `hasDeferredFragments` to the `RootFieldBuilder.Result`. We could keep track of this as a `var` on the `RootFieldBuilder`, and while building, if we encounter a deferred fragment, we just set that to `true`.